servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletHttpRequest.java

```diff
@@ -168,7 +170,11 @@ public <T> Optional<T> get(CharSequence name, ArgumentConversionContext<T> conve
                 Objects.requireNonNull(name, NULL_KEY);
                 Object attribute = null;
                 try {
-                    attribute = delegate.getAttribute(name.toString());
+                    if (asyncContext != null) {
```

Comment:
extract this delegation logic into a `delegate()` method

Reply:
or alternatively, just reassign the `delegate` field when initializing the asyncContext.